dpm/client/__init__.py

```diff
@@ -160,10 +160,15 @@ def publish(self, publisher=None):
             self._upload_file(path, filedata[path])
 
         # TODO: (?) echo('Finalizing ... ', nl=False)
+        data_package_s3_url = filedata['datapackage.json']['upload_url']
         response = self._apirequest(
             method='POST',
-            url='/api/package/%s/%s/finalize' % (self.username, self.datapackage.descriptor['name'])
+            url='/api/package/upload',
+            json={'datapackage': data_package_s3_url}
         )
+        status = response.json().get('status', None)
+        if status is None or status != 'queued':
+            raise DpmException('server did not provide upload authorization for files')
```

Comment:
Strictly i think the error message is something like: "Unable to finalize publication of the data package. Data is in bitstore but not published."

However for now I think this is ok.